frontend/src/pages/modelServing/screens/projects/InferenceServiceModal/DataConnectionFolderPathField.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import { FormGroup, InputGroup, InputGroupText, TextInput } from '@patternfly/react-core';
+
+type DataConnectionFolderPathFieldProps = {
+  folderPath: string;
+  setFolderPath: (folderPath: string) => void;
+};
+
+const DataConnectionFolderPathField: React.FC<DataConnectionFolderPathFieldProps> = ({
+  folderPath,
+  setFolderPath,
+}) => {
+  const [error, setError] = React.useState('');
+
+  const formatLeadingSlash = (value: string) => {
+    if (folderPath === '/') {
+      return '';
+    }
+    return value;
+  };
+  return (
+    <FormGroup
+      helperTextInvalid={error}
+      validated={error ? 'error' : 'default'}
+      fieldId="folder-path"
+      label="Folder path"
+    >
+      <InputGroup>
+        <InputGroupText variant="plain">/</InputGroupText>
+        <TextInput
+          aria-label="folder-path"
+          type="text"
+          value={formatLeadingSlash(folderPath)}
+          placeholder="eg. data"
+          onChange={(value) => {
+            const cleanedPath = value.replace(/^\/+/, '');
+            let error = '';
+            if (!/^[a-zA-Z0-9.\-_*'()/]+$/.test(cleanedPath)) {
```

Comment:
Looks like there is something wrong with this testing regex, when the input is empty, it will throw an error. It won't happen for the first time (when you open the modal and don't input anything in the path). However, the error will show when you input something and delete them all. We should allow the user to leave the input empty if they want to mount it on the root path.
<img width="540" alt="Screenshot 2023-03-28 at 11 08 16 AM" src="https://user-images.githubusercontent.com/37624318/228119081-a35fa622-e795-44ba-bbb9-b89c46265d32.png">